source/scripts/ContentScript/inject/stargazerProvider.txt.ts

```diff
@@ -67,106 +71,126 @@ async function handleRequest(chain, req) {
 
   // All request accounts go to the same place
   if (method === 'requestAccounts') {
-    const { result, data } = await window.providerManager.enable();
-
-    if (!result) throw ERRORS.USER_REJECTED()
+    const {result, data} = await window.providerManager.enable();
 
+    if (!result) {
+      throw ERRORS.USER_REJECTED();
+    }
     return data.accounts;
   }
 
-  if (
-    prefix &&
-    REQUEST_MAP.hasOwnProperty(prefix) &&
-    REQUEST_MAP[prefix].hasOwnProperty(method)
-  ) {
-    return provider.getMethod(REQUEST_MAP[prefix][method])(...req.params);
+  if (prefix && REQUEST_MAP.hasOwnProperty(prefix) && REQUEST_MAP[prefix].hasOwnProperty(method)) {
+    const response = await provider.getMethod(REQUEST_MAP[prefix][method])(... req.params);
+
+    if (response.result === false) {
+      throw ERRORS.USER_REJECTED();
+    }
+
+    return response;
   }
 
   throw ERRORS.INVALID_METHOD();
 }
 
-const provider = (chain) => {
+const provider = chain => {
   chain = chain.toLowerCase();
-  
+
   if (!Object.keys(SUPPORTED_CHAINS).includes(chain)) {
     console.error('Unsupported chain: ' + chain);
   }
 
   return {
-    request: async (req) => {
-      const params = req.params || []
+    request: async req => {
+      const params = req.params || [];
+
       return await handleRequest(chain, {
         method: req.method,
         params
+      }).catch( err => {
+        console.error(err);
+        throw err;
```

Comment:
add on this catch hanndling so we can console.error the error. 
Metamask seems to be doing somethign similar with an error middleware